lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -441,173 +510,272 @@ def create_autoscaling_group(connection, module):
-                 min_size=min_size,
-                 max_size=max_size,
-                 placement_group=placement_group,
-                 desired_capacity=desired_capacity,
-                 vpc_zone_identifier=vpc_zone_identifier,
-                 connection=connection,
-                 tags=asg_tags,
-                 health_check_period=health_check_period,
-                 health_check_type=health_check_type,
-                 default_cooldown=default_cooldown,
-                 termination_policies=termination_policies)
+        launch_configs = connection.describe_launch_configurations(LaunchConfigurationNames=[launch_config_name])
+        ag = dict(
+            AutoScalingGroupName=group_name,
+            AvailabilityZones=availability_zones,
+            LaunchConfigurationName=launch_configs['LaunchConfigurations'][0]['LaunchConfigurationName'],
+            MinSize=min_size,
+            MaxSize=max_size,
+            DesiredCapacity=desired_capacity,
+            VPCZoneIdentifier=vpc_zone_identifier,
+            Tags=asg_tags,
+            HealthCheckGracePeriod=health_check_period,
+            HealthCheckType=health_check_type,
+            DefaultCooldown=default_cooldown,
+            TerminationPolicies=termination_policies)
+        if placement_group:
+            ag['PlacementGroup'] = placement_group
+        if load_balancers:
+            ag['LoadBalancerNames'] = load_balancers
+        if target_group_arns:
+            ag['TargetGroupARNs'] = target_group_arns
 
         try:
-            connection.create_auto_scaling_group(ag)
-            suspend_processes(ag, module)
+            connection.create_auto_scaling_group(**ag)
+            ag = connection.describe_auto_scaling_groups(
+                AutoScalingGroupNames=[group_name])['AutoScalingGroups'][0]
+            suspend_processes(connection, ag, module)
             if wait_for_instances:
                 wait_for_new_inst(module, connection, group_name, wait_timeout, desired_capacity, 'viable_instances')
-                wait_for_elb(connection, module, group_name)
-
+                if load_balancers:
+                    wait_for_elb(connection, module, group_name)
+                # Wait for target group health if target group(s)defined
+                if target_group_arns:
+                    wait_for_target_group(connection, module, group_name)
             if notification_topic:
-                ag.put_notification_configuration(notification_topic, notification_types)
-
-            as_group = connection.get_all_groups(names=[group_name])[0]
+                connection.put_notification_configuration(
+                    AutoScalingGroupName=group_name,
+                    TopicARN=notification_topic,
+                    NotificationTypes=[
+                        notification_types,
+                    ]
+                )
+            as_group = connection.describe_auto_scaling_groups(
```

Comment:
It looks like this call and the above (line 570) call are filling 2 variables with the same information. Is there a reason two names are used?